extension/radia.js

```diff
@@ -0,0 +1,97 @@
+const axios = require('axios').default;
+
+async function listen(nodecg) {
+	const casters = nodecg.Replicant('casters');
+	const radiaSettings = nodecg.Replicant('radiaSettings');
+	var APIURL;
+	var Authentication;
+
+	if (
+		!nodecg.bundleConfig ||
+		typeof nodecg.bundleConfig.radia === 'undefined'
+	) {
+		nodecg.log.error(
+			`"radia" is not defined in cfg/${nodecg.bundleName}.json! The ability to import data via the Radia
+			Production API is not possible`
+		);
+		radiaSettings.value.enabled = false;
+		return;  // Exit from extension as keys aren't available to use.
+	} else {
+		APIURL = nodecg.bundleConfig.radia.url;
+		Authentication = nodecg.bundleConfig.radia.Authentication;
+		radiaSettings.value.enabled = true;
+	}
+
+	nodecg.listenFor('getLiveCommentators', async (data, ack) => {
+		getLiveCasters(APIURL, Authentication, radiaSettings.value.guildID)
+			.then((data) => {
+				// Clear all currently stored caster data
+				for (const [key, value] of Object.entries(casters.value)){
+					delete casters.value[key];
+				}
+				// Get how many casters we need to place into the replicant
+				var numberOfCaster = 3;
+				if(data.length < numberOfCaster){
+					numberOfCaster = data.length;
+				}
+				// Place data into replicant
+				for(var i = 0; i < numberOfCaster; i++){
+					casters.value[generateId()] = data.pop();
+				}
+				ack(null, data);  // We return any casters left in the array after taking our max of 3
+			})
+			.catch((err) => {
+				// If the API gives us a 404, just ignore it :)
+				if(err.response.status === 404){
+					ack(null, null);
+				}
+				ack(err);
+			});
+	});
+}
+
+/**
+ * Generate a random ID
+ * @returns {string}
+ */
+function generateId() {
+	return '' + Math.random().toString(36).substr(2, 9);
```

Comment:
This is duplicated across multiple files (even mine...), it'd be best to include this in a single utility file.